proto/decode.go

```diff
@@ -492,11 +494,22 @@ func (o *Buffer) unmarshalType(st reflect.Type, prop *StructProperties, is_group
 			// Maybe it's an extension?
 			if prop.extendable {
 				if e, _ := extendable(structPointer_Interface(base, st)); isExtensionField(e, int32(tag)) {
-					if err = o.skip(st, tag, wire); err == nil {
-						extmap := e.extensionsWrite()
-						ext := extmap[int32(tag)] // may be missing
-						ext.enc = append(ext.enc, o.buf[oi:o.index]...)
-						extmap[int32(tag)] = ext
+					if !regExtInit {
+						msgType := reflect.Zero(reflect.PtrTo(st)).Interface().(Message)
+						regExt = RegisteredExtensions(msgType)
```

Comment:
Maybe it's more clear to just do `regExt := RegisteredExtensions(msgType)` here. But I'm weary of doing any work more than necessary in something like proto unmarshaling since that is likely to be a hot path in many servers. (Though, admittedly, extensions are likely used infrequently enough that this is not necessary. Suggestions?)